transports/rabbitmq/delayed-delivery.md

```diff
@@ -50,10 +56,43 @@ end
 
 The delay levels are connected in this manner, from highest (27) to lowest (0). Each delay level's routing key's add wildcards as needed so that they are looking at the portion of the message's routing key that corresponds to its delay level.
 
+The following table illustrates dead letter exchange and bindings for the delay queues. It's important to note the wildcard rules for RabbitMQ binding expressions, where *word* describes a dot-delimited segment:
+
+* `*` substitutes for exactly one word
+* `#` substitutes for 0 or more words
+
+| Exchange & Queue Name | x-dead-letter-exchange | Queue Binding                                               |
+|-----------------------|------------------------|-------------------------------------------------------------|
+| nsb.delay-level-27    | nsb.delay-level-26     | `1.#`                                                       |
+| nsb.delay-level-26    | nsb.delay-level-25     | `*.1.#`                                                     |
+| nsb.delay-level-25    | nsb.delay-level-24     | `*.*.1.#`                                                   |
+| ...                   | ...                    | ...                                                         |
+| nsb.delay-level-03    | nsb.delay-level-02     | `*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.1.#`       |
+| nsb.delay-level-02    | nsb.delay-level-01     | `*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.1.#`     |
+| nsb.delay-level-01    | nsb.delay-level-00     | `*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.1.#`   |
+| nsb.delay-level-00    | nsb.delay-delivery     | `*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.1.#` |
+
+With this system of exchanges and queues, the message is first delivered to the first applicable queue based on the binary value. In the 10-second example (binary `1010`) the message would first be delivered to the `nsb.delay-level-03` exchange, where the matching queue's binding will only match if a `1` exists in the 4th-to-last digit. In that case the queue will "accept" the message into the `nsb.delay-level-03` queue and wait for something to process it up until the TTL (2^3 seconds) before forwarding to the `nsb.delay-level-02` exchange.
```

Comment:
Probably should drop the first "first": "the message is delivered to the first applicable"

Reply:
Also, that should be "first applicable exchange" not queue. Messages are never sent to queues!

Reply:
"In that case the queue will "accept" the message into the `nsb.delay-level-03`"

Messages are published to an exchange, and the all of the bindings to the exchange are evaluated, and all ones that are valid get a copy of the message sent to the destination of the binding. 

So, try to rewrite this from the perspective of the exchange, not the queue. Message is published to `nsb.delay-level-03`, which has two bindings, the 0 binding and 1 binding. If the 1 binding is a match, then the message is delivered to the queue. If the 0 binding matches, the message gets delivered to the `nsb.delay-level-02` exchange.

Reply:
"before forwarding to the `nsb.delay-level-02` exchange."

This should point out that the TTL expiring causes the message to dead-lettered, so it is republished to the configured exchange, `nsb.delay-level-02` in this case.